### PR TITLE
[New Scheduler]Add CreationJobManager

### PR DESCRIPTION
**Pull request by jiangpengcheng**
## Description

This module records container creation using a `CreationId`, when a new container creation request is created, one entry with an unique `CreationId` will be created and saved in to ETCD, and will be deleted when creation is finished(or timeout waiting for creation complete ack).

So that a `MemoryQueue` can query how many containers are being created from ETCD and then do containers scheduling based on this value.

It can also help to trace/debug the procedure of a container creation.

## Related issue and scope

- [ ] I opened an issue to propose and discuss this change (#????)

## My changes affect the following components

- [ ] API
- [ ] Controller
- [ ] Message Bus (e.g., Kafka)
- [ ] Loadbalancer
- [x] Scheduler
- [ ] Invoker
- [ ] Intrinsic actions (e.g., sequences, conductors)
- [ ] Data stores (e.g., CouchDB)
- [ ] Tests
- [ ] Deployment
- [ ] CLI
- [ ] General tooling
- [ ] Documentation

## Types of changes

- [ ] Bug fix (generally a non-breaking change which closes an issue).
- [x] Enhancement or new feature (adds new functionality).
- [ ] Breaking change (a bug fix or enhancement which changes existing behavior).

## Checklist:

- [x] I signed an [Apache CLA](https://github.com/apache/openwhisk/blob/master/CONTRIBUTING.md).
- [x] I reviewed the [style guides](https://github.com/apache/openwhisk/wiki/Contributing:-Git-guidelines#code-readiness) and followed the recommendations (Travis CI will check :).
- [x] I added tests to cover my changes.
- [ ] My changes require further changes to the documentation.
- [ ] I updated the documentation where necessary.

### CODE REVIEW

File: core/scheduler/src/main/scala/org/apache/openwhisk/core/scheduler/container/CreationJobManager.scala [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+                ContainerCreationError.TimeoutError,
+                s"timeout waiting for the ack of $creationId after $timeout")))
+      dataManagementService ! UnregisterData(
+        inProgressContainer(invocationNamespace, action, revision, schedulerInstanceId, creationId))
+    }
+  }
+
+  private val topicPrefix = loadConfigOrThrow[String](ConfigKeys.kafkaTopicsPrefix)
+  private val topic = s"${topicPrefix}creationAck${schedulerInstanceId.asString}"
+  private val maxActiveAcksPerPoll = 128
```
Comment (ningyougang) [inline core/scheduler/src/main/scala/org/apache/openwhisk/core/scheduler/container/CreationJobManager.scala:240]: In Scheduler.scala, this value is configurable.
Can we make it configurable as well?

File: core/scheduler/src/main/scala/org/apache/openwhisk/core/scheduler/container/CreationJobManager.scala [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+case object GetPoolStatus
+
+case class JobEntry(action: FullyQualifiedEntityName, timer: Cancellable)
+
+class CreationJobManager(feedFactory: (ActorRefFactory, String, Int, Array[Byte] => Future[Unit]) => ActorRef,
+                         schedulerInstanceId: SchedulerInstanceId,
+                         dataManagementService: ActorRef)(implicit actorSystem: ActorSystem, logging: Logging)
+    extends Actor {
+  private implicit val ec: ExecutionContext = actorSystem.dispatcher
+  private val baseTimeout = loadConfigOrThrow[Int](ConfigKeys.schedulerInProgressJobRetentionSecond).seconds
```
Comment (ningyougang) [inline core/scheduler/src/main/scala/org/apache/openwhisk/core/scheduler/container/CreationJobManager.scala:90]: For time configuration, it is better to change it to `loadConfigOrThrow[FiniteDuration]`

**Comment by ningyougang** [comment]
LGTM with some small suggestions

File: core/scheduler/src/main/scala/org/apache/openwhisk/core/scheduler/container/CreationJobManager.scala (hunk 1 above)
Comment (jiangpengcheng) [reply core/scheduler/src/main/scala/org/apache/openwhisk/core/scheduler/container/CreationJobManager.scala:240]: it seems other places are also hardcoded

File: core/scheduler/src/main/scala/org/apache/openwhisk/core/scheduler/container/CreationJobManager.scala (hunk 2 above)
Comment (jiangpengcheng) [reply core/scheduler/src/main/scala/org/apache/openwhisk/core/scheduler/container/CreationJobManager.scala:90]: great catch, I will update

**Review by ningyougang — APPROVED** [review]

**Review by style95 — APPROVED** [review]

**Review by KeonHee — APPROVED** [review]

**Pull request merged**